src/async-syntax-highlighter.js

```diff
@@ -15,41 +17,84 @@ export default (options) => {
       return ReactAsyncHighlighter.loadAstGenerator();
     }
 
-    static registerLanguage = (_, language) => {
-      if (!supportsLanguageRegistering) {
+    static async loadLanguage(language) {
+      const languageLoader = languageLoaders[language];
+      if (typeof languageLoader === 'function') {
+        return languageLoader(ReactAsyncHighlighter.registerLanguage);
+      } else {
+        throw `Language ${language} not supported`
+      }
+    }
+    
+    static isRegistered = (language) => {
+      if(!registerLanguage) {
+        return true;
+      }
+
+      if (!ReactAsyncHighlighter.astGenerator) {
+        // Ast generator not available yet, but language will be registered once it is.
+        return ReactAsyncHighlighter.languages.findIndex(item => item.name === language) > -1;
+      }
+
+      return isLanguageRegistered(ReactAsyncHighlighter.astGenerator, language);
+    }
+
+    static registerLanguage = (name, language) => {
+      if (!registerLanguage) {
         return;
       }
       
       if(ReactAsyncHighlighter.astGenerator) {
-        return ReactAsyncHighlighter.astGenerator.register(language);
+        return registerLanguage(ReactAsyncHighlighter.astGenerator, name, language);
       } else {
-        ReactAsyncHighlighter.languages.push(language);
+        ReactAsyncHighlighter.languages.push({
```

Comment:
can we use map with concated string key? then lookups will be O(1)?

Reply:
Done